pkg/receive/limiter.go

```diff
@@ -5,92 +5,42 @@ package receive
 
 import (
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/thanos-io/thanos/pkg/extprom"
+	"github.com/thanos-io/thanos/pkg/gate"
 )
 
-const (
-	seriesLimitName    = "series"
-	samplesLimitName   = "samples"
-	sizeBytesLimitName = "body_size"
-)
-
-type requestLimiter struct {
-	sizeBytesLimit   int64
-	seriesLimit      int64
-	samplesLimit     int64
-	limitsHit        *prometheus.SummaryVec
-	configuredLimits *prometheus.GaugeVec
+type limiter struct {
+	requestLimiter requestLimiter
+	writeGate      gate.Gate
+	// activeSeriesLimiter *activeSeriesLimiter `yaml:"active_series"`
```

Comment:
Do we need these commented struct fields?
I can add an active series limit to this config in a follow-up PR, maybe a `TODO` would be better here? 🙂

Reply:
Fixed in https://github.com/thanos-io/thanos/pull/5565/commits/e5c171d29e2651d7f49f5c3acc55bb87d8074421. 👍